api/client.ts

```diff
@@ -422,7 +433,7 @@ export class Transport {
       if (this._debug) {
         console.debug("close:", evt.code, evt.reason);
       }
-      this._closeCallback(evt.code);
+      this._closeCallback(toError(evt.code.toString()));
```

Comment:
It looks like the type of `_closeCallback` is incorrect. I think we need a new type

```typescript
type CloseCallback = (code: number) => void;

...

  _closeCallback: CloseCallback;
```

Then we need to update each place that's using _closeCallback to have a callback of type `CloseCallback`.

```typescript
logout(callback?: CloseCallback) {
....
close(callback?: (code: number, callback: CloseCallback) => void) {
```

etc.

Reply:
Thanks for the review! I think the type for `logout` `callback` should also be `(code: number, callback: CloseCallback) => void)` instead of `CloseCallback`.

Reply:
Ah yes, you're right!